### PR TITLE
Proposal for new repository for UI work

### PR DESCRIPTION
**Pull request by vishnoianil**

### CODE REVIEW

**Comment by vishnoianil** [comment]
FYI @nerdalert @Gregory-Pereira @renrenstart @juancappi @mingxzhao

File: docs/ui/project-proposal.md
Comment (leseb) [inline docs/ui/project-proposal.md:11]:
```suggestion
Currently we don’t have public repository under InstrucLab that hosts UI-related work. The goal of this project is to provide a space for hacking a UI for the InstructLab. The scope of this project is to support the following personas through the UI:
```

**Review by leseb — CHANGES_REQUESTED** [review]
Is the plan to transfer the content of `instructlab/instructlab-bot` (UI directory) over to the newly proposed repository?

Or are we starting from scratch? Thanks!

File: docs/ui/project-proposal.md
Comment (vishnoianil) [reply docs/ui/project-proposal.md]: Fixed.

File: docs/ui/project-proposal.md
Comment (vishnoianil) [reply docs/ui/project-proposal.md]: Thanks for catching this @leseb .Fixed

File: docs/ui/project-proposal.md
Comment (vishnoianil) [reply docs/ui/project-proposal.md]: Thanks for the review @leseb . Fixed

**Review by Gregory-Pereira — APPROVED** [review]
Great job on this, short and to the point
/lgtm

**Comment by vishnoianil** [comment]
> Is the plan to transfer the content of `instructlab/instructlab-bot` (UI directory) over to the newly proposed repository?
> 
> Or are we starting from scratch? Thanks!

@leseb Yes, plan it to use the existing code from `instructlab/instructlab-bot/ui` as a seed code.

File: docs/ui/project-proposal.md
Comment (russellb) [inline docs/ui/project-proposal.md:14]: Could you clarify if you would only focus on the upstream project workflow, or if you would be creating something people could use to help with managing their own taxonomy repos?  Is it both?

Comment (vishnoianil) [reply docs/ui/project-proposal.md:14]: @russellb Main focus of this project is for the upstream workflow (for upstream contributors and upstream traigers). Intent is to develop it as a SaaS service and deploy it somewhere on Cloud (ROSA, IBM Cloud). But that doesn't stop users from deploying this stack in their private data center to manage their own private taxonomy repo if they want to follow similar workflow downstream.

Comment (vishnoianil) [reply docs/ui/project-proposal.md:14]: @russellb Let me know if you have any follow-up questions, otherwise I will push the patch to update the doc.

Comment (russellb) [reply docs/ui/project-proposal.md:14]: I think something designed to be reused is important, too, so I just want to make it really clear whether this project is aiming to serve that need for instructlab or if it's explicitly an upstream-oriented SaaS only.

I think you're saying

- build something deployed as a SaaS for the upstream project and its taxonomy repo and process
- build in a way that someone could deploy the same service on their own infrastructure for managing their own taxonomy repo

Is that right?

Comment (vishnoianil) [reply docs/ui/project-proposal.md:14]: Yes, thats correct.

Comment (vishnoianil) [reply docs/ui/project-proposal.md:14]: Done. Thanks for the thorough review @russellb

**Review by russellb — COMMENTED** [review]
lgtm - going to wait for more reviews and feedback first before merging though

**Comment by jjasghar** [comment]
:shipit: says the squirrel.

File: docs/ui/project-proposal.md
Comment (nathan-weinberg) [inline docs/ui/project-proposal.md:30]: Could we following the scheme of having a GitHub Team - say `UI Maintainers` - with these folks as the initial members?

**Review by nathan-weinberg — CHANGES_REQUESTED** [review]

File: docs/ui/project-proposal.md
Comment (vishnoianil) [reply docs/ui/project-proposal.md:30]: @nathan-weinberg I think we need the proposal to be first approved to go ahead and make the github team, otherwise it doesn't make sense to do that?. I personally would prefer to list the name of the initial maintainers, just to capture the diversity of the collaborators. But happy to fix it according to any policy we have.

Comment (russellb) [reply docs/ui/project-proposal.md:30]: I think the suggestion is just to make your first sentence

> The initial team of maintainers (UI Maintainers) for this repository will be:

Comment (nathan-weinberg) [reply docs/ui/project-proposal.md:30]: Yeah the idea is that after the proposal is merged we could create said `UI Maintainers` GitHub Team and put all y'all on it - tighter permissions-wise and easier for adding/removing people later

Comment (vishnoianil) [reply docs/ui/project-proposal.md]: Got it. Fixed it. Thanks @russellb @nathan-weinberg for clarifying.

**Comment by leseb** [comment]
> > Is the plan to transfer the content of `instructlab/instructlab-bot` (UI directory) over to the newly proposed repository?
> > Or are we starting from scratch? Thanks!
> 
> @leseb Yes, plan it to use the existing code from `instructlab/instructlab-bot/ui` as a seed code.

Please make it clear in the doc, thanks! Other than this, this LGTM.

**Review by lhawthorn — APPROVED** [review]
LGTM - nice work @vishnoianil

**Review by nathan-weinberg — APPROVED** [review]

**Comment by vishnoianil** [comment]
> > > Is the plan to transfer the content of `instructlab/instructlab-bot` (UI directory) over to the newly proposed repository?
> > > Or are we starting from scratch? Thanks!
> > 
> > 
> > @leseb Yes, plan it to use the existing code from `instructlab/instructlab-bot/ui` as a seed code.
> 
> Please make it clear in the doc, thanks! Other than this, this LGTM.

@leseb Done, updated the doc, PTAL.

**Review by leseb — APPROVED** [review]

**Comment by leseb** [comment]
Whoever has the right to create the new repo can go ahead and get this merged, thanks! @russellb is that you?

**Review by russellb — APPROVED** [review]
There seems to be consensus on this and a majority of the O.C. has approved.

I will now:

- [x] merge
- [x] create the repo
- [ ] Create the UI Maintainers team
- [ ] give that team access to the repo

**Comment by russellb** [comment]
> There seems to be consensus on this and a majority of the O.C. has approved.
> 
> I will now:
> 
> * [x]  merge
> * [x]  create the repo
> * [x]  Create the UI Maintainers team
> * [x]  give that team access to the repo

This is done except I have not added everyone to the UI maintainers team yet. I will do that if someone can get me the associated github usernames.

**Pull request merged**
